bin/navsnmp

```diff
@@ -69,13 +69,14 @@ def get_matching_boxes(patterns):
 
 
 def snmp_printer(netbox):
-    version = ''
-    if netbox.snmp_version == 1:
-        version = '1'
-    elif netbox.snmp_version == 2:
-        version = '2c'
+    profile = netbox.get_preferred_snmp_management_profile(writeable=True)
+    version = (
+        profile.snmp_version
+        if getattr(profile, "snmp_version", None) in ('1', '2c')
+        else ''
+    )
```

Comment:
Nah, this still doesn't work. Partly because I wasn't accurate in my statements, but you also didn't attempt to verify them 😆 

If you check with existing management profiles, the `snmp_version` field is actually either the string `1` or the string `2`. `2c` isn't used. However, `2c` is the string that  is expected by SNMP command line tools.

Reply:
Also, still not sure why you need to use `getattr` here. AFAICT, `snmp_version` is defined as a property on the `ManagementProfile` class, so it will always be there.